Day-II-Calculator/calculator/src/App.js

```diff
@@ -3,104 +3,100 @@ import './App.css';
 import NumberButton from './components/ButtonComponents/NumberButton';
 import ActionButton from './components/ButtonComponents/ActionButton';
 import CalculatorDisplay from './components/DisplayComponents/CalculatorDisplay';
+import calculator from './calculatorData';
 
-const calculator = [
-  {
-    value: 'clear',
-    style: 'ActionButton text',
-    type: 'action'
-  },
-  {
-    value: '÷',
-    style: 'NumberButton red',
-    type: 'number'
-  },
-  {
-    value: 7,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: 8,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: 9,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: 'X',
-    style: 'NumberButton red',
-    type: 'number'
-  },
-  {
-    value: 4,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: 5,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: 6,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: '–',
-    style: 'NumberButton red',
-    type: 'number'
-  },
-  {
-    value: 1,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: 2,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: 3,
-    style: 'NumberButton',
-    type: 'number'
-  },
-  {
-    value: '+',
-    style: 'NumberButton red',
-    type: 'number'
-  },
-  {
-    value: 0,
-    style: 'ActionButton text',
-    type: 'action'
-  },
-  {
-    value: '=',
-    style: 'NumberButton red',
-    type: 'number'
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      total: 0,
+      displayValue: '0',
+      operatorSelected: false
+    };
   }
-];
 
-const App = () => {
-  return (
-    <div className="App">
-      <CalculatorDisplay style="CalculatorDisplay" value={0} />
+  add(num) {
+    this.setState(prevState => {
+      return {
+        total: prevState.total + num,
+        displayValue: this.state.total
```

Comment:
use prevState for this, not this.state (this is how you do it with a function rather than an object for the argument for this.setState)